adl/common/http.adl

```diff
@@ -2,35 +2,19 @@ module common.http {
 
 /// Request types
 
-struct HttpGet<O> {
+struct HttpReq<I,O> {
+  HttpMethod method;
```

Comment:
I'd perhaps set a default of "post" here, indicating that "get" should  only be chosen when confident that the call is side-effect free (or otherwise capable of being cached by default)